docs/kubernetes/development.md

```diff
@@ -1,60 +1,100 @@
 # Kubernetes Development
 
-## Manual
+## Manual deployment
 
 To build and install a development version of the driver:
 
-```
-$ GCE_PD_CSI_STAGING_IMAGE=gcr.io/path/to/driver/image:dev   # Location to push dev image to
-$ make push-container
+```sh
+GCE_PD_CSI_STAGING_IMAGE=gcr.io/path/to/driver/image:dev   # Location to push dev image to
+make push-container
 
 # Modify controller.yaml and node.yaml in ./deploy/kubernetes/dev to use dev image
-$ GCE_PD_DRIVER_VERSION=dev
-$ ./deploy/kubernetes/deploy-driver.sh
+GCE_PD_DRIVER_VERSION=dev
+./deploy/kubernetes/deploy-driver.sh
 ```
 
 To bring down driver:
 
-```
-$ ./deploy/kubernetes/delete-driver.sh
+```sh
+./deploy/kubernetes/delete-driver.sh
 ```
 ## Testing
 
-### E2E Tests:
+### E2E tests:
+
+The following environment variables will be required for setup and all subsequent test runs.
+
+```sh
+export PROJECT=my-project                                            # GCP project used for testing
+export GCE_PD_SA_NAME=$PROJECT-pd-sa                                 # Service account name
+export GCE_PD_SA_DIR=/my/safe/credentials/directory                  # Service account directory
+export GOOGLE_APPLICATION_CREDENTIALS=$GCE_PD_SA_DIR/cloud-sa.json   # Service account key
+```
 
 #### One time setup
 
-```console
-$ export PROJECT=my-project                               # GCP Project to run tests in
-$ export GCE_PD_SA_NAME=$PROJECT-pd-sa                 # Name of the service account to create
-$ export GCE_PD_SA_DIR=/my/safe/credentials/directory    # Directory to save the service account key
-$ export ENABLE_KMS=false
-$ ./deploy/setup-project.sh
+The following steps will need to be completed once per project.
+
+##### Create service account
+
+To create a service account with correct permissions, your test project must:
+1. Be assigned [Project IAM Admin](https://cloud.google.com/iam/docs/understanding-roles#resourcemanager.projectIamAdmin) and [Role Administrator](https://cloud.google.com/iam/docs/understanding-roles#iam.roleAdmin) IAM roles.
+2. Be part of an organization that allows creation of service account keys.
+   - Google developer projects restrict SA keys and will not work for E2E testing.
+
+```sh
+export ENABLE_KMS=false
+./deploy/setup-project.sh
 ```
-#### Ongoing runs
-```console
-$ export PROJECT=my-project                               # GCP Project to run tests in
-$ export GCE_PD_SA_NAME=$PROJECT-pd-sa
-$ export IAM_NAME=$GCE_PD_SA_NAME@$PROJECT.iam.gserviceaccount.com  # IAM SA that was set up in "one time setup"
```

Comment:
Do we still need this IAM_NAME? I think we still need this for the setup project script?

Reply:
Yes, we do need this. Nice catch.